test/built-ins/Atomics/wait/negative-index-throws.js

```diff
@@ -0,0 +1,21 @@
+// Copyright (C) 2018 Amal Hussein. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+/*---
+esid: sec-map-objects
+description: Ensure that null BufferData typedArray param must be an object TODO
+info: |
+  Atomics.wait( typedArray, index, value, timeout )
+
+  2.Let i be ? ValidateAtomicAccess(typedArray, index).
+    ...
+      2.Let accessIndex be ? ToIndex(requestIndex).
+        ...
+        2.b If integerIndex < 0, throw a RangeError exception
+features: [ TODO ]
+---*/
+
+var sab = new SharedArrayBuffer(1024);
+var int32Array = new Int32Array(sab);
+
+assert.throws(RangeError, () => Atomics.wait(int32Array, -300, 0, 0));
+assert.throws(RangeError, () => Atomics.wait(int32Array, -0.999, 0, 0));
```

Comment:
in addition (or instead of these as you prefer) you might wanna try some extreme cases: `-1`, `-0.000001`, `-Infinity`.

In the test that checks for a valid index, you may use `-0` to confirm it's valid and treated as `0`.

Reply:
you may also poison the next arguments here as well: 

```js
var poisoned = {
  get valueOf() { throw new Test262Error("Should not evaluate this code"); }
};
assert.throws(RangeError, function() {
  Atomics.wait(int32Array, -1, poisoned, poisoned);
});
```

so you assert `index` is verified before `value`